deployment/ccip/changeset/solana_v0_1_1/cs_idl.go

```diff
@@ -672,3 +662,161 @@ func UpgradeIDL(e cldf.Environment, c IDLConfig) (cldf.ChangesetOutput, error) {
 
 	return cldf.ChangesetOutput{}, nil
 }
+
+// Build instruction to interact with Anchor IDL using the list of ids above for each message
+func buildIdlInstruction(programID solana.PublicKey, accountsForIx solana.AccountMetaSlice, idlInstruction int, params []byte) (solana.GenericInstruction, error) {
+	data := binary.LittleEndian.AppendUint64([]byte{}, IdlIxTag) // 8-byte Extend instruction identifier
+	data = append(data, byte(idlInstruction))                    // Append the numeric ID of the operation
+	data = append(data, params...)                               // Append any additional parameters
+
+	instruction := solana.NewInstruction(
+		programID,
+		accountsForIx,
+		data,
+	)
+	return *instruction, nil
+}
+
+func calculateAuthority(e cldf.Environment, c IDLConfig) (solana.PublicKey, error) {
+	timelockSignerPDA, err := FetchTimelockSigner(e, c.ChainSelector)
+	if err != nil {
+		return solana.PublicKey{}, fmt.Errorf("error loading timelockSignerPDA: %w", err)
+	}
+	authority := e.BlockChains.SolanaChains()[c.ChainSelector].DeployerKey.PublicKey()
+	if c.MCMS != nil {
+		authority = timelockSignerPDA
+	}
+	return authority, err
```

Comment:
Nitpick, this can be simplified
```suggestion
	if c.MCMS != nil {
		return FetchTimelockSigner(e, c.ChainSelector)
	}
	return e.BlockChains.SolanaChains()[c.ChainSelector].DeployerKey.PublicKey(),  err
```